web/thesauruses/_meta/io.json

```diff
@@ -0,0 +1,243 @@
+{
+    "meta": {
+        "language": "language_id",
+        "language_version": "version.number",
+        "language_name": "Human-Friendly Language Name"
+    },
+    "categories": {
+        "Console IO": [
+            "write_line",
+            "write_line_with_new_line",
+            "read_line",
+            "read_char",
+            "clear_console_cutput",
+            "change_console_color"
+        ],
+        "File IO": [
+            "list_directory",
+            "read_directory_info",
+            "create_directory",
+            "delete_directory",
+            "rename_directory",
+            "move_directory",
+            "update_directory_permissions",
+            "read_line_of_file",
+            "read_lines_of_file",
+            "read_file_stream",
+            "create_file",
+            "delete_file",
+            "update_file",
+            "move_file"
+        ],
+        "Network IO": [
+            "send_a_get_request",
+            "send_a_post_request",
+            "add_headers",
+            "add_query_params",
+            "inbuilt_url_class",
+            "open_a_socket",
+            "close_a_socket",
+            "add_authorization_headers",
+            "add_cookies",
+            "read_cookies",
+            "read_headers"
+        ],
+        "Database IO": [
+            "open_connection",
+            "close_connection",
+            "execute_query_single_result_set",
+            "execute_query_multiple_result_sets",
+            "execute_non_query"
+        ],
+        "JSON": [
+            "serialize_json",
+            "deserialize_json",
+            "serialize_json_with_options",
+            "deserialize_json_with_options"
+        ],
+        "XML": [
+            "serialize_xml",
+            "deserialize_xml",
+            "serialize_xml_with_options",
+            "deserialize_xml_with_options"
+        ]
```

Comment:
This is probably a good idea! Maybe add CSV too? (CSV tends to have a ton of options too, but maybe we can keep it simpler?)

Reply:
I did think about CSV however unlike JSON and XML there doesn't tend to be a inbuild language or framework feature to parse them at least in my experience. Again happy to add this in if you think it is valuable however.

Reply:
Python has the `csv` built-in library, but I'm not sure about others off the top of my head. I realize it could be more complicated due to the dozen different ways to set them up, but we could just keep it simple.